File: flutter/lib/widgets/create_post_modal.dart
import 'package:flutter/material.dart';
import 'package:font_awesome_flutter/font_awesome_flutter.dart';
import 'dart:ui';
import '../theme/app_theme.dart';
import 'primary_button.dart';
import 'glass_button.dart';

/// Instagram-style post creation modal
class CreatePostModal extends StatefulWidget {
  final Function(Map<String, dynamic>)? onPost;

  const CreatePostModal({
    Key? key,
    this.onPost,
  }) : super(key: key);

  @override
  State<CreatePostModal> createState() => _CreatePostModalState();

  static Future<T?> show<T>({
    required BuildContext context,
    Function(Map<String, dynamic>)? onPost,
  }) {
    return showModalBottomSheet<T>(
      context: context,
      isScrollControlled: true,
      backgroundColor: Colors.transparent,
      barrierColor: Colors.black.withOpacity(0.7),
      builder: (context) => CreatePostModal(onPost: onPost),
    );
  }
}

class _CreatePostModalState extends State<CreatePostModal>
    with SingleTickerProviderStateMixin {
  late AnimationController _controller;
  late Animation<Offset> _slideAnimation;

  final TextEditingController _captionController = TextEditingController();
  final TextEditingController _locationController = TextEditingController();

  String _selectedCategory = 'Meets & Events';
  final List<String> _categories = [
    'Meets & Events',
    'Modifications',
    'Track Days',
    'For Sale',
    'General',
  ];

  List<String> _selectedImages = [];
  bool _isPublic = true;

  @override
  void initState() {
    super.initState();
    _controller = AnimationController(
      duration: const Duration(milliseconds: 600),
      vsync: this,
    );

    _slideAnimation = Tween<Offset>(
      begin: const Offset(0, 1),
      end: Offset.zero,
    ).animate(
      CurvedAnimation(
        parent: _controller,
        curve: Curves.elasticOut,
      ),
    );

    _controller.forward();
  }

  @override
  void dispose() {
    _controller.dispose();
    _captionController.dispose();
    _locationController.dispose();
    super.dispose();
  }

  void _handleClose() {
    _controller.reverse().then((_) {
      Navigator.of(context).pop();
    });
  }

  void _handlePost() {
    final postData = {
      'caption': _captionController.text,
      'location': _locationController.text,
      'category': _selectedCategory,
      'images': _selectedImages,
      'isPublic': _isPublic,
      'timestamp': DateTime.now(),
    };

    widget.onPost?.call(postData);
    _handleClose();
  }

  void _addImage() {
    // Simulate image selection
    setState(() {
      _selectedImages.add('image_${_selectedImages.length + 1}.jpg');
    });
  }

  void _removeImage(int index) {
    setState(() {
      _selectedImages.removeAt(index);
    });
  }

  @override
  Widget build(BuildContext context) {
    final isDark = Theme.of(context).brightness == Brightness.dark;

    return SlideTransition(
      position: _slideAnimation,
      child: DraggableScrollableSheet(
        initialChildSize: 0.9,
        minChildSize: 0.5,
        maxChildSize: 0.95,
        builder: (context, scrollController) {
          return Container(
            decoration: BoxDecoration(
              borderRadius: const BorderRadius.only(
                topLeft: Radius.circular(24),
                topRight: Radius.circular(24),
              ),
              boxShadow: [
                BoxShadow(
                  color: Colors.black.withOpacity(0.3),
                  blurRadius: 30,
                  offset: const Offset(0, -10),
                ),
              ],
            ),
            child: ClipRRect(
              borderRadius: const BorderRadius.only(
                topLeft: Radius.circular(24),
                topRight: Radius.circular(24),
              ),
              child: BackdropFilter(
                filter: ImageFilter.blur(sigmaX: 20, sigmaY: 20),
                child: Container(
                  decoration: BoxDecoration(
                    gradient: LinearGradient(
                      begin: Alignment.topLeft,
                      end: Alignment.bottomRight,
                      colors: isDark
                          ? [
                              const Color(0xFF1C1C1E).withOpacity(0.95),
                              const Color(0xFF2C2C2E).withOpacity(0.95),
                            ]
                          : [
                              Colors.white.withOpacity(0.95),
                              Colors.grey.shade50.withOpacity(0.95),
                            ],
                    ),
                    border: Border(
                      top: BorderSide(
                        color: Colors.white.withOpacity(0.2),
                        width: 1.5,
                      ),
                    ),
                  ),
                  child: Column(
                    children: [
                      // Header
                      _buildHeader(isDark),

                      // Content
                      Expanded(
                        child: ListView(
                          controller: scrollController,
                          padding: const EdgeInsets.all(20),
                          children: [
                            // Image Section
                            _buildImageSection(isDark),
                            const SizedBox(height: 20),

                            // Caption
                            _buildCaptionField(isDark),
                            const SizedBox(height: 16),

                            // Location
                            _buildLocationField(isDark),
                            const SizedBox(height: 16),

                            // Category
                            _buildCategorySection(isDark),
                            const SizedBox(height: 16),

                            // Privacy Toggle
                            _buildPrivacyToggle(isDark),
                            const SizedBox(height: 24),

                            // Post Button
                            PrimaryButton(
                              text: 'Share Post',
                              icon: FontAwesomeIcons.share,
                              onPressed: _handlePost,
                              height: 54,
                            ),
                            const SizedBox(height: 12),
                            GlassButton(
                              text: 'Save as Draft',
                              icon: FontAwesomeIcons.floppyDisk,
                              onPressed: () {},
                            ),
                          ],
                        ),
                      ),
                    ],
                  ),
                ),
              ),
            ),
          );
        },
      ),
    );
  }

  Widget _buildHeader(bool isDark) {
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 16),
      decoration: BoxDecoration(
        border: Border(
          bottom: BorderSide(
            color: isDark
                ? Colors.white.withOpacity(0.1)
                : Colors.black.withOpacity(0.1),
            width: 1,
          ),
        ),
      ),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          GestureDetector(
            onTap: _handleClose,
            child: Icon(
              Icons.close,
              color: isDark ? Colors.white70 : Colors.black54,
              size: 28,
            ),
          ),
          Text(
            'New Post',
            style: TextStyle(
              fontSize: 20,
              fontWeight: FontWeight.w700,
              color: isDark ? Colors.white : Colors.black87,
            ),
          ),
          const SizedBox(width: 28),
        ],
      ),
    );
  }

  Widget _buildImageSection(bool isDark) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          'Photos',
          style: TextStyle(
            fontSize: 16,
            fontWeight: FontWeight.w600,
            color: isDark ? Colors.white : Colors.black87,
          ),
        ),
        const SizedBox(height: 12),
        SizedBox(
          height: 120,
          child: ListView(
            scrollDirection: Axis.horizontal,
            children: [
              // Add Image Button
              GestureDetector(
                onTap: _addImage,
                child: Container(
                  width: 120,
                  height: 120,
                  margin: const EdgeInsets.only(right: 12),
                  decoration: BoxDecoration(
                    color: isDark
                        ? Colors.white.withOpacity(0.05)
                        : Colors.grey.shade200,
                    borderRadius: BorderRadius.circular(16),
                    border: Border.all(
                      color: AppTheme.primaryBlue.withOpacity(0.3),
                      width: 2,
                      style: BorderStyle.solid,
                    ),
                  ),
                  child: Column(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: [
                      FaIcon(
                        FontAwesomeIcons.plus,
                        color: AppTheme.primaryBlue,
                        size: 32,
                      ),
                      const SizedBox(height: 8),
                      Text(
                        'Add Photo',
                        style: TextStyle(
                          fontSize: 12,
                          color: AppTheme.primaryBlue,
                          fontWeight: FontWeight.w600,
                        ),
                      ),
                    ],
                  ),
                ),
              ),

              // Selected Images
              ..._selectedImages.asMap().entries.map((entry) {
                final index = entry.key;
                final image = entry.value;
                return Stack(
                  children: [
                    Container(
                      width: 120,
                      height: 120,
                      margin: const EdgeInsets.only(right: 12),
                      decoration: BoxDecoration(
                        color: AppTheme.primaryBlue.withOpacity(0.2),
                        borderRadius: BorderRadius.circular(16),
                      ),
                      child: Center(
                        child: FaIcon(
                          FontAwesomeIcons.image,
                          color: AppTheme.primaryBlue.withOpacity(0.5),
                          size: 32,
                        ),
                      ),
                    ),
                    Positioned(
                      top: 8,
                      right: 20,
                      child: GestureDetector(
                        onTap: () => _removeImage(index),
                        child: Container(
                          width: 28,
                          height: 28,
                          decoration: BoxDecoration(
                            color: Colors.red,
                            shape: BoxShape.circle,
                            boxShadow: [
                              BoxShadow(
                                color: Colors.black.withOpacity(0.3),
                                blurRadius: 8,
                              ),
                            ],
                          ),
                          child: const Icon(
                            Icons.close,
                            color: Colors.white,
                            size: 16,
                          ),
                        ),
                      ),
                    ),
                  ],
                );
              }).toList(),
            ],
          ),
        ),
      ],
    );
  }

  Widget _buildCaptionField(bool isDark) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          'Caption',
          style: TextStyle(
            fontSize: 16,
            fontWeight: FontWeight.w600,
            color: isDark ? Colors.white : Colors.black87,
          ),
        ),
        const SizedBox(height: 8),
        TextField(
          controller: _captionController,
          maxLines: 4,
          style: TextStyle(
            color: isDark ? Colors.white : Colors.black87,
            fontSize: 15,
          ),
          decoration: InputDecoration(
            hintText: 'Write a caption...',
            hintStyle: TextStyle(
              color: isDark ? Colors.white60 : Colors.black45,
            ),
            filled: true,
            fillColor: isDark
                ? Colors.white.withOpacity(0.05)
                : Colors.white.withOpacity(0.5),
            border: OutlineInputBorder(
              borderRadius: BorderRadius.circular(14),
              borderSide: BorderSide(
                color: Colors.white.withOpacity(0.2),
              ),
            ),
            enabledBorder: OutlineInputBorder(
              borderRadius: BorderRadius.circular(14),
              borderSide: BorderSide(
                color: Colors.white.withOpacity(0.2),
              ),
            ),
            focusedBorder: OutlineInputBorder(
              borderRadius: BorderRadius.circular(14),
              borderSide: BorderSide(
                color: AppTheme.primaryBlue,
                width: 2,
              ),
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildLocationField(bool isDark) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Row(
          children: [
            Icon(
              Icons.location_on,
              size: 18,
              color: AppTheme.primaryBlue,
            ),
            const SizedBox(width: 6),
            Text(
              'Location',
              style: TextStyle(
                fontSize: 16,
                fontWeight: FontWeight.w600,
                color: isDark ? Colors.white : Colors.black87,
              ),
            ),
          ],
        ),
        const SizedBox(height: 8),
        TextField(
          controller: _locationController,
          style: TextStyle(
            color: isDark ? Colors.white : Colors.black87,
            fontSize: 15,
          ),
          decoration: InputDecoration(
            hintText: 'Add location',
            hintStyle: TextStyle(
              color: isDark ? Colors.white60 : Colors.black45,
            ),
            filled: true,
            fillColor: isDark
                ? Colors.white.withOpacity(0.05)
                : Colors.white.withOpacity(0.5),
            border: OutlineInputBorder(
              borderRadius: BorderRadius.circular(14),
              borderSide: BorderSide(
                color: Colors.white.withOpacity(0.2),
              ),
            ),
            enabledBorder: OutlineInputBorder(
              borderRadius: BorderRadius.circular(14),
              borderSide: BorderSide(
                color: Colors.white.withOpacity(0.2),
              ),
            ),
            focusedBorder: OutlineInputBorder(
              borderRadius: BorderRadius.circular(14),
              borderSide: BorderSide(
                color: AppTheme.primaryBlue,
                width: 2,
              ),
            ),
            prefixIcon: Icon(
              FontAwesomeIcons.locationDot,
              size: 18,
              color: isDark ? Colors.white60 : Colors.black45,
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildCategorySection(bool isDark) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          'Category',
          style: TextStyle(
            fontSize: 16,
            fontWeight: FontWeight.w600,
            color: isDark ? Colors.white : Colors.black87,
          ),
        ),
        const SizedBox(height: 12),
        Wrap(
          spacing: 8,
          runSpacing: 8,
          children: _categories.map((category) {
            final isSelected = _selectedCategory == category;
            return GestureDetector(
              onTap: () {
                setState(() => _selectedCategory = category);
              },
              child: Container(
                padding: const EdgeInsets.symmetric(
                  horizontal: 16,
                  vertical: 10,
                ),
                decoration: BoxDecoration(
                  gradient: isSelected ? AppTheme.primaryGradient : null,
                  color: isSelected
                      ? null
                      : (isDark
                          ? Colors.white.withOpacity(0.05)
                          : Colors.white.withOpacity(0.5)),
                  borderRadius: BorderRadius.circular(20),
                  border: Border.all(
                    color: isSelected
                        ? AppTheme.primaryBlue
                        : Colors.white.withOpacity(0.2),
                    width: isSelected ? 2 : 1,
                  ),
                ),
                child: Text(
                  category,
                  style: TextStyle(
                    fontSize: 13,
                    fontWeight: isSelected ? FontWeight.w600 : FontWeight.w400,
                    color: isSelected
                        ? Colors.white
                        : (isDark ? Colors.white70 : Colors.black87),
                  ),
                ),
              ),
            );
          }).toList(),
        ),
      ],
    );
  }

  Widget _buildPrivacyToggle(bool isDark) {
    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: isDark
            ? Colors.white.withOpacity(0.05)
            : Colors.white.withOpacity(0.5),
        borderRadius: BorderRadius.circular(14),
        border: Border.all(
          color: Colors.white.withOpacity(0.2),
        ),
      ),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Row(
            children: [
              Icon(
                _isPublic ? FontAwesomeIcons.globe : FontAwesomeIcons.lock,
                size: 20,
                color: AppTheme.primaryBlue,
              ),
              const SizedBox(width: 12),
              Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    _isPublic ? 'Public' : 'Private',
                    style: TextStyle(
                      fontSize: 15,
                      fontWeight: FontWeight.w600,
                      color: isDark ? Colors.white : Colors.black87,
                    ),
                  ),
                  Text(
                    _isPublic ? 'Everyone can see' : 'Only followers',
                    style: TextStyle(
                      fontSize: 12,
                      color: isDark ? Colors.white60 : Colors.black54,
                    ),
                  ),
                ],
              ),
            ],
          ),
          Switch(
            value: _isPublic,
            onChanged: (value) {
              setState(() => _isPublic = value);
            },
            activeColor: AppTheme.primaryBlue,
          ),
        ],
      ),
    );
  }
}

File: react-native/src/components/CreatePostModal.js
import React, { useState, useRef } from 'react';
import {
  View,
  Text,
  StyleSheet,
  Modal,
  TouchableOpacity,
  TextInput,
  ScrollView,
  Animated,
  useColorScheme,
  Switch,
} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import Icon from 'react-native-vector-icons/FontAwesome5';
import AppTheme from '../theme/AppTheme';
import PrimaryButton from './PrimaryButton';
import GlassButton from './GlassButton';

const CreatePostModal = ({ visible, onClose, onPost }) => {
  const isDark = useColorScheme() === 'dark';
  const [caption, setCaption] = useState('');
  const [location, setLocation] = useState('');
  const [selectedCategory, setSelectedCategory] = useState('Meets & Events');
  const [selectedImages, setSelectedImages] = useState([]);
  const [isPublic, setIsPublic] = useState(true);

  const slideAnim = useRef(new Animated.Value(0)).current;

  const categories = [
    'Meets & Events',
    'Modifications',
    'Track Days',
    'For Sale',
    'General',
  ];

  React.useEffect(() => {
    if (visible) {
      Animated.spring(slideAnim, {
        toValue: 1,
        friction: 8,
        useNativeDriver: true,
      }).start();
    } else {
      Animated.timing(slideAnim, {
        toValue: 0,
        duration: 200,
        useNativeDriver: true,
      }).start();
    }
  }, [visible]);

  const handleAddImage = () => {
    setSelectedImages([...selectedImages, `image_${selectedImages.length + 1}.jpg`]);
  };

  const handleRemoveImage = (index) => {
    const newImages = [...selectedImages];
    newImages.splice(index, 1);
    setSelectedImages(newImages);
  };

  const handlePost = () => {
    const postData = {
      caption,
      location,
      category: selectedCategory,
      images: selectedImages,
      isPublic,
      timestamp: new Date(),
    };
    onPost?.(postData);
    onClose();
    // Reset form
    setCaption('');
    setLocation('');
    setSelectedImages([]);
  };

  const translateY = slideAnim.interpolate({
    inputRange: [0, 1],
    outputRange: [600, 0],
  });

  return (
    <Modal
      visible={visible}
      animationType="fade"
      transparent
      onRequestClose={onClose}>
      <View style={styles.overlay}>
        <Animated.View
          style={[
            styles.modalContainer,
            { transform: [{ translateY }] },
          ]}>
          <LinearGradient
            colors={
              isDark
                ? ['rgba(28, 28, 30, 0.95)', 'rgba(44, 44, 46, 0.95)']
                : ['rgba(255, 255, 255, 0.95)', 'rgba(245, 245, 245, 0.95)']
            }
            style={styles.modalContent}>
            {/* Header */}
            <View style={[styles.header, {
              borderBottomColor: isDark ? 'rgba(255, 255, 255, 0.1)' : 'rgba(0, 0, 0, 0.1)',
            }]}>
              <TouchableOpacity onPress={onClose}>
                <Icon name="times" size={28} color={isDark ? '#FFFFFFB3' : '#00000099'} />
              </TouchableOpacity>
              <Text style={[styles.headerTitle, { color: isDark ? '#FFF' : '#000' }]}>
                New Post
              </Text>
              <View style={{ width: 28 }} />
            </View>

            {/* Content */}
            <ScrollView style={styles.scrollView} showsVerticalScrollIndicator={false}>
              {/* Images Section */}
              <View style={styles.section}>
                <Text style={[styles.sectionTitle, { color: isDark ? '#FFF' : '#000' }]}>
                  Photos
                </Text>
                <ScrollView horizontal showsHorizontalScrollIndicator={false}>
                  {/* Add Image Button */}
                  <TouchableOpacity
                    style={[styles.addImageButton, {
                      backgroundColor: isDark ? 'rgba(255, 255, 255, 0.05)' : '#E0E0E0',
                      borderColor: `${AppTheme.colors.primaryBlue}50`,
                    }]}
                    onPress={handleAddImage}>
                    <Icon name="plus" size={32} color={AppTheme.colors.primaryBlue} />
                    <Text style={[styles.addImageText, { color: AppTheme.colors.primaryBlue }]}>
                      Add Photo
                    </Text>
                  </TouchableOpacity>

                  {/* Selected Images */}
                  {selectedImages.map((image, index) => (
                    <View key={index} style={styles.imageContainer}>
                      <View style={[styles.imagePlaceholder, {
                        backgroundColor: `${AppTheme.colors.primaryBlue}30`,
                      }]}>
                        <Icon
                          name="image"
                          size={32}
                          color={`${AppTheme.colors.primaryBlue}80`}
                        />
                      </View>
                      <TouchableOpacity
                        style={styles.removeButton}
                        onPress={() => handleRemoveImage(index)}>
                        <Icon name="times" size={16} color="#FFFFFF" />
                      </TouchableOpacity>
                    </View>
                  ))}
                </ScrollView>
              </View>

              {/* Caption */}
              <View style={styles.section}>
                <Text style={[styles.sectionTitle, { color: isDark ? '#FFF' : '#000' }]}>
                  Caption
                </Text>
                <TextInput
                  style={[styles.textInput, {
                    backgroundColor: isDark ? 'rgba(255, 255, 255, 0.05)' : 'rgba(255, 255, 255, 0.5)',
                    borderColor: 'rgba(255, 255, 255, 0.2)',
                    color: isDark ? '#FFF' : '#000',
                  }]}
                  placeholder="Write a caption..."
                  placeholderTextColor={isDark ? '#FFFFFF99' : '#00000099'}
                  value={caption}
                  onChangeText={setCaption}
                  multiline
                  numberOfLines={4}
                  textAlignVertical="top"
                />
              </View>

              {/* Location */}
              <View style={styles.section}>
                <View style={styles.locationHeader}>
                  <Icon name="map-marker-alt" size={18} color={AppTheme.colors.primaryBlue} />
                  <Text style={[styles.sectionTitle, { marginLeft: 6, marginBottom: 0, color: isDark ? '#FFF' : '#000' }]}>
                    Location
                  </Text>
                </View>
                <TextInput
                  style={[styles.textInput, {
                    backgroundColor: isDark ? 'rgba(255, 255, 255, 0.05)' : 'rgba(255, 255, 255, 0.5)',
                    borderColor: 'rgba(255, 255, 255, 0.2)',
                    color: isDark ? '#FFF' : '#000',
                  }]}
                  placeholder="Add location"
                  placeholderTextColor={isDark ? '#FFFFFF99' : '#00000099'}
                  value={location}
                  onChangeText={setLocation}
                />
              </View>

              {/* Category */}
              <View style={styles.section}>
                <Text style={[styles.sectionTitle, { color: isDark ? '#FFF' : '#000' }]}>
                  Category
                </Text>
                <View style={styles.categoriesContainer}>
                  {categories.map((category) => {
                    const isSelected = selectedCategory === category;
                    return (
                      <TouchableOpacity
                        key={category}
                        style={[
                          styles.categoryChip,
                          {
                            backgroundColor: isSelected
                              ? 'transparent'
                              : (isDark ? 'rgba(255, 255, 255, 0.05)' : 'rgba(255, 255, 255, 0.5)'),
                            borderColor: isSelected
                              ? AppTheme.colors.primaryBlue
                              : 'rgba(255, 255, 255, 0.2)',
                            borderWidth: isSelected ? 2 : 1,
                          },
                        ]}
                        onPress={() => setSelectedCategory(category)}>
                        {isSelected && (
                          <LinearGradient
                            colors={AppTheme.gradients.primary}
                            style={StyleSheet.absoluteFillObject}
                          />
                        )}
                        <Text
                          style={[
                            styles.categoryText,
                            {
                              color: isSelected
                                ? '#FFFFFF'
                                : (isDark ? '#FFFFFFB3' : '#000'),
                              fontWeight: isSelected ? '600' : '400',
                            },
                          ]}>
                          {category}
                        </Text>
                      </TouchableOpacity>
                    );
                  })}
                </View>
              </View>

              {/* Privacy Toggle */}
              <View style={[styles.privacyContainer, {
                backgroundColor: isDark ? 'rgba(255, 255, 255, 0.05)' : 'rgba(255, 255, 255, 0.5)',
                borderColor: 'rgba(255, 255, 255, 0.2)',
              }]}>
                <View style={styles.privacyInfo}>
                  <Icon
                    name={isPublic ? 'globe' : 'lock'}
                    size={20}
                    color={AppTheme.colors.primaryBlue}
                  />
                  <View style={styles.privacyText}>
                    <Text style={[styles.privacyTitle, { color: isDark ? '#FFF' : '#000' }]}>
                      {isPublic ? 'Public' : 'Private'}
                    </Text>
                    <Text style={[styles.privacySubtitle, { color: isDark ? '#FFFFFF99' : '#00000099' }]}>
                      {isPublic ? 'Everyone can see' : 'Only followers'}
                    </Text>
                  </View>
                </View>
                <Switch
                  value={isPublic}
                  onValueChange={setIsPublic}
                  trackColor={{ true: AppTheme.colors.primaryBlue, false: '#999' }}
                />
              </View>

              {/* Action Buttons */}
              <View style={styles.actionButtons}>
                <PrimaryButton
                  text="Share Post"
                  icon="share"
                  onPress={handlePost}
                  height={54}
                  style={styles.shareButton}
                />
                <GlassButton
                  text="Save as Draft"
                  icon="save"
                  onPress={() => {}}
                  width="100%"
                  height={50}
                  style={styles.draftButton}
                />
              </View>

              <View style={{ height: 40 }} />
            </ScrollView>
          </LinearGradient>
        </Animated.View>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  overlay: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.7)',
    justifyContent: 'flex-end',
  },
  modalContainer: {
    height: '90%',
    borderTopLeftRadius: 24,
    borderTopRightRadius: 24,
    overflow: 'hidden',
  },
  modalContent: {
    flex: 1,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 20,
    borderBottomWidth: 1,
  },
  headerTitle: {
    fontSize: 20,
    fontWeight: '700',
  },
  scrollView: {
    flex: 1,
    padding: 20,
  },
  section: {
    marginBottom: 20,
  },
  sectionTitle: {
    fontSize: 16,
    fontWeight: '600',
    marginBottom: 12,
  },
  addImageButton: {
    width: 120,
    height: 120,
    borderRadius: 16,
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 12,
    borderWidth: 2,
    borderStyle: 'dashed',
  },
  addImageText: {
    fontSize: 12,
    fontWeight: '600',
    marginTop: 8,
  },
  imageContainer: {
    width: 120,
    height: 120,
    marginRight: 12,
  },
  imagePlaceholder: {
    flex: 1,
    borderRadius: 16,
    justifyContent: 'center',
    alignItems: 'center',
  },
  removeButton: {
    position: 'absolute',
    top: 8,
    right: 8,
    width: 28,
    height: 28,
    borderRadius: 14,
    backgroundColor: '#FF3B30',
    justifyContent: 'center',
    alignItems: 'center',
    ...AppTheme.shadows.card,
  },
  textInput: {
    borderRadius: 14,
    borderWidth: 1,
    padding: 12,
    fontSize: 15,
    minHeight: 100,
  },
  locationHeader: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 12,
  },
  categoriesContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  categoryChip: {
    paddingHorizontal: 16,
    paddingVertical: 10,
    borderRadius: 20,
    marginRight: 8,
    marginBottom: 8,
    overflow: 'hidden',
  },
  categoryText: {
    fontSize: 13,
  },
  privacyContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 16,
    borderRadius: 14,
    borderWidth: 1,
    marginBottom: 24,
  },
  privacyInfo: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  privacyText: {
    marginLeft: 12,
  },
  privacyTitle: {
    fontSize: 15,
    fontWeight: '600',
  },
  privacySubtitle: {
    fontSize: 12,
  },
  actionButtons: {
    marginTop: 4,
  },
  shareButton: {
    marginBottom: 12,
  },
  draftButton: {
    marginBottom: 12,
  },
});

export default CreatePostModal;
